### ISSUE TITLE
Upon resize of the native container, the Flutter contents momentarily "flash" as they are redrawn

### ISSUE THREAD

**Issue by mehmetf** [flutter/flutter#19189]
If the FlutterView on Android is resized, the contents of the view disappear and reappear. This momentary redraw causes a flash on the screen. Coupled with the black background of the Flutter view, it is quite unappealing. Opening this issue as a discussion on whether there's anything we can do about it.

**Comment by eseidelGoogle** [comment]
I wonder what Chrome does.  Since Chrome must have this problem when resizing the window (and having to asynchronously communicate that to the rendering process).

**Comment by eseidelGoogle** [comment]
Presumably we could continue drawing the previous frame after redraw (which the next frame is prepared) if that's preferred.  Curious for your thoughts @cbracken @chinmaygarde

**Comment by cbracken** [comment]
That seems like a reasonable initial approach to get rid of the flickering.

/cc @jason-simmons who may have been looking at something similar recently.

**Comment by matthew-carroll** [comment]
I just took a look at this.  I created a test Activity that reduces the height of a FlutterView by 50% when a button is pressed, and then back.

By default, I do see a very short-lived black flicker.

What I just did was give my FlutterView a transparent background and I noticed that it solved the flicker issue.  This probably isn't a good solution in general because the transparent rendering is probably expensive, and may only be relevant for the few cases where a FlutterView is changing size.

As far as any solution you guys are envisioning, would it be a C-side change?  Or would it be something in on the Android side?  I suppose we might be able to go transparent just before a resize and then return back, but I'm not sure if that's actually feasible given that we don't really control the nuances of frame rendering timing.

Either way, I think right now is the time to figure out what we want to do about this.

**Comment by mehmetf** [comment]
- Do we have the capability to convert view background to transparent on iOS?
- I would imagine you still see a flicker in the form of content disappearing and reappearing? (Even though it is not as jarring as a black box momentarily blinking).

**Comment by matthew-carroll** [comment]
Yes, the content disappears and reappears, though it happens very quickly.  I'm not sure what, if anything, can be done about that.

I'll defer to @dnfield on the iOS transparency question.

**Comment by mehmetf** [comment]
Thanks Matt. I would think the real workaround is to set the view background the same color as the Flutter background. Otherwise, even transparent background does not work if native background does not match Flutter background (you will see a flash of native background color). Imagine a native background of light gray and a Flutter card with white background that is resizing. You would see a flash of gray I assume. In this case, you don't want transparent, you want white. Could you confirm this?

If the content disappearing and reappearing is very brief, then I believe we can close this issue. Let's see how happy the customers are. If even that becomes a problem then I would like to have a solution similar to what's proposed by @eseidelGoogle above where we continue to draw what was last seen on the view until the new frame is ready.

**Comment by matthew-carroll** [comment]
@mehmetf I'm thinking we can't really facilitate the situation you're describing in general.  Often backgrounds are gradients or patterns.  I don't think we can effectively expose control over that to help with this use-case, especially considering that said gradient or pattern will have ambiguous behavior when changing size.

I would look at this more like any other Android View.  If we change the size of a ListView, RecyclerView, or even just a regular View with a background color, how does that look?  I think that should be a reasonable goal for a FlutterView, right?

**Comment by mehmetf** [comment]
You are right. If the Flutter view being resized has a non-trivial background then resizing will cause artifacts. I think we can live with that for now.

Your experiment is interesting. I never tried setting a non-trivial background to an Android view and resizing it. I am sure you can implement it in a nice way (by coding the transition) but the default behavior is likely an abrupt shift.
